Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,114 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
+
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"HTTP-ClientError"];
+    sentryException.mechanism =
+        [[SentryMechanism alloc] initWithType:@"SentryNetworkTrackingIntegration"];
+
+    if (threads.count > 0) {
+        for (SentryThread *thread in threads) {
+            if (nil != thread.current && [thread.current boolValue]) {
+                SentryStacktrace *sentryStacktrace = [thread stacktrace];
+                sentryStacktrace.snapshot = @(YES);
+
+                sentryException.stacktrace = sentryStacktrace;
+
+                break;
+            }
+        }
+    }
+
+    SentryRequest *request = [[SentryRequest alloc] init];
+
+    NSURLRequest *myRequest = (NSURLRequest *)sessionTask.currentRequest;
+
+    NSURL *url = [[sessionTask currentRequest] URL];
+
+    NSString *urlString = url.absoluteString;
+
+    if (nil != url.fragment) {
+        request.fragment = url.fragment;
+        urlString = [urlString
+            stringByReplacingOccurrencesOfString:[@"#" stringByAppendingString:url.fragment]
+                                      withString:@""];
+    }
+
+    if (nil != url.query) {
+        request.queryString = url.query;
+        urlString =
+            [urlString stringByReplacingOccurrencesOfString:[@"?" stringByAppendingString:url.query]
+                                                 withString:@""];
+    }
+
+    request.url = urlString;
+    request.method = myRequest.HTTPMethod;
+    if (sessionTask.countOfBytesSent != 0) {
+        request.bodySize = [NSNumber numberWithLongLong:sessionTask.countOfBytesSent];
+    }
+    if (nil != myRequest.allHTTPHeaderFields) {
+        NSDictionary<NSString *, NSString *> *headers = myRequest.allHTTPHeaderFields.copy;
+        request.headers = headers;
+        request.cookies = headers[@"Cookie"];
+    }
+
+    event.exceptions = @[ sentryException ];
+    event.request = request;
+
+    NSMutableDictionary<NSString *, id> *context = [[NSMutableDictionary alloc] init];
+    ;
+    NSMutableDictionary<NSString *, id> *response = [[NSMutableDictionary alloc] init];
+
+    [response setValue:responseStatusCode forKey:@"status_code"];
+    if (nil != myResponse.allHeaderFields) {
+        NSDictionary<NSString *, NSString *> *headers = myResponse.allHeaderFields.copy;
+        [response setValue:headers forKey:@"headers"];
+        [response setValue:headers[@"Cookie"] forKey:@"cookies"];
+    }
+    if (sessionTask.countOfBytesReceived != 0) {
+        [response setValue:[NSNumber numberWithLongLong:sessionTask.countOfBytesReceived]
+                    forKey:@"body_size"];
+    }
+
+    context[@"response"] = response;
+    event.context = context;
+
+    [SentrySDK captureEvent:event];
+}
+
+- (BOOL)containsStatusCode:(NSInteger)statusCode
+{
+    for (SentryHttpStatusCodeRange *range in SentrySDK.options.failedRequestStatusCodes) {
```

Comment:
`m`: As pointed out already above, I would pass in all required options as dependencies when calling the constructor.

Reply:
I agree, just don't understand why this one isn't fine and the trace propagation targets, can you clarify?

Reply:
Just a matter of preference for testability. I vote for passing everything in as a parameter—no strong opinion.

Reply:
I understand the motivation and I agree, my question was why the other property was not a ctor param and it was ok.
As we agreed, we will refactor this in another PR.